models/models.go

```diff
@@ -12,10 +13,32 @@ import (
 	"github.com/18F/cf-cdn-service-broker/utils"
 )
 
+type State string
+
+const (
+	Provisioning   State = "provisioning"
+	Provisioned          = "provisioned"
+	Deprovisioning       = "deprovisioning"
+	Deprovisioned        = "deprovisioned"
+)
+
+func (s State) Value() (driver.Value, error) {
+	return string(s), nil
+}
+
+func (s *State) Scan(value interface{}) error {
+	bytes, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("error scanning status %s", value)
+	}
+	*s = State(bytes)
+	return nil
+}
```

Comment:
This is waaay over my head in Go-land...mind explaining what these two methods are doing, just for my edification?